s3fs/core.py

```diff
@@ -142,7 +169,10 @@ def _ls(self, path, refresh=False):
                     f['Size'] = 0
                     del f['Name']
             else:
-                files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                try:
+                    files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                except ClientError:
+                    files = []
```

Comment:
I would appreciate your help, in that case, on making a thorough set of conditions and test for `ls` - I found it a bit tricky to decide what it should do, is it always the same as `glob(path)`?